### ISSUE TITLE
Object.extname is deprecated.

### ISSUE THREAD

**Issue by fireice009** [Glavin001/atom-beautify#1423]
Atom beta 1.13.0-beta8
atom-beautify 0.29.14

Remove and then install, the issue is still there.

Argument to `path.extname` must be a string
```
Object.extname (/Users/my/Downloads/Atom Beta.app/Contents/Resources/app.asar/src/electron-shims.js:20:10)
<unknown> (/Users/my/.atom/packages/atom-beautify/src/beautifiers/index.coffee:230:34)
Promise._execute (/Users/my/.atom/packages/atom-beautify/node_modules/bluebird/js/release/debuggability.js:299:9)
Promise._resolveFromExecutor (/Users/my/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:481:18)
new Promise (/Users/my/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:77:14)
<unknown> (/Users/my/.atom/packages/atom-beautify/src/beautifiers/index.coffee:226:18)
```

**Comment by Glavin001** [comment]
This is a result of https://github.com/atom/atom/commit/c876fd76f8daf7c3bd5932cdb488f17980197698 and https://github.com/atom/atom/pull/12300

**Comment by ftaiolivista** [comment]
Same for me. Atom 1.13.1 x64 on Linux

**Comment by steronydh** [comment]
Same here for Atom 1.14 and atom-beautify 0.29,17.

**Comment by tmdNinja** [comment]
same for atom 1.13.1 on mac

**Comment by fedorovstas** [comment]
Same for me. Atom 1.13.1 Ubuntu 16.04.

**Comment by viralk** [comment]
Same for me: Atom 1.14.2, Windows 10 Pro, atom-beautify 0.29.17

**Comment by jaimemunera** [comment]
Same here: Atom v1.14.3 Mac OS (Sierra), atom-beautify 0.29.17

**Comment by kengggg** [comment]
Same here: Atom v1.14.3 macOS Sierra

**Comment by xtianF** [comment]
same-same el cap mac

**Comment by d4nirod** [comment]
Fixed updating to 1.14.4, no more deprecation warnings.

**Comment by shillem** [comment]
Same here: atom 1.14.4 with macOS Sierra

**Comment by rishikrsharma** [comment]
same here  Atom 1.14.4 with macOS Sierra 
(Argument to path.extname must be a string)

**Comment by Olmer1st** [comment]
same for atom 1.15.0 Ubuntu 15.04

**Comment by FadhelC** [comment]
Same here Atom 1.15, Mac OS X (El Capitan)

**Comment by ShadowFM** [comment]
Same here.

Atom v. 1.16.0-beta0
Windows 10

**Comment by Arcanemagus** [comment]
@Glavin001 although the commits you reference are what is causing this to be shown to users, the actual problem is that `filePath` isn't being verified to have a value before it is used in a call to `path.extname` [here](https://github.com/Glavin001/atom-beautify/blob/cfe0e6ad8e5227e018e10bc0703d7aa5fd6cc9c1/src/beautifiers/index.coffee#L245) (likely source of this), or [here](https://github.com/Glavin001/atom-beautify/blob/cfe0e6ad8e5227e018e10bc0703d7aa5fd6cc9c1/src/beautifiers/index.coffee#L145).

**Comment by Israaael** [comment]
Hi, I'm new. Not sure if this has been resolved or not. I'm having the same problem. W 10 pro x64, Atom 1.14.3 and beautify 0.29.17.

**Comment by dramaticsquirrel** [comment]
Atom 1.15.0 ia32
atom-beautify 0.29.17

Windows 10 Home

**Comment by Glavin001** [comment]
Great eye @Arcanemagus! There is an open Pull Request being reviewed with should resolve this issue once it is merged: https://github.com/Glavin001/atom-beautify/pull/1573
Thank you all for your patience!

**Comment by Glavin001** [comment]
Published to v0.29.18
